packages/ui/components/formbuilder/testUtils.ts

```diff
@@ -0,0 +1,184 @@
+  addFieldLabel: "Add Field",
+  disabled: false,
+  LockedIcon: false,
+  dataStore: { options: {} },
+  disableOnPrefill: false,
+};
+
+export const mockFieldProps: FormBuildFieldProps = {
+  field: {
+    type: "text",
+    name: "test-text-n",
+    variant: "text",
+    placeholder: "text-test-p",
+    required: false,
+    label: "text-test-l",
+    disableOnPrefill: false,
+    value: "",
+  },
+  readOnly: false,
+};
+
+export const setMockMatchMedia = () => {
+  Object.defineProperty(window, "matchMedia", {
+    value: vi.fn().mockImplementation((query) => ({
+      matches: false,
+      media: query,
+      onchange: null,
+      addEventListener: vi.fn(),
+      removeEventListener: vi.fn(),
+      dispatchEvent: vi.fn(),
+    })),
+  });
+};
+
+export const setMockIntersectionObserver = () => {
+  Object.defineProperty(window, "IntersectionObserver", {
+    value: vi.fn().mockImplementation(() => ({
+      observe: vi.fn(),
+      unobserve: vi.fn(),
+      disconnect: vi.fn(),
+      takeRecords: vi.fn(() => []),
+    })),
+  });
+};
+
+export const clickDisablePrefillBasedOnUserInput = (disableOnPrefill?: boolean) => {
+  if (typeof disableOnPrefill === "boolean") {
+    // Get the checkbox element
+    const checkbox = getDisablePrefillCheckbox(disableOnPrefill) as HTMLInputElement;
+
+    // Determine if the checkbox's current state differs from the `disableOnPrefill` prop
+    const isChecked = checkbox ? checkbox.checked : false;
+
+    // Click the checkbox only if its current state differs from the `disableOnPrefill` value
+    if (isChecked !== disableOnPrefill) {
+      fireEvent.click(checkbox);
+    }
+  }
+};
+
+export const getDisablePrefillCheckbox = (disableOnPrefill?: boolean) => {
+  if (typeof disableOnPrefill === "boolean") {
+    const checkbox = screen.getByRole("checkbox", { name: /disable_input_if_prefilled/i });
+    return checkbox;
+  }
+};
+
+export const checkForDisablePreFillCheckox = (disableOnPrefill?: boolean) => {
+  const checkbox = getDisablePrefillCheckbox(disableOnPrefill);
+  expect(checkbox).toBeInTheDocument();
+};
+
+export const questionUtils = {
+  addQuestion: async (props: QuestionProps) => {
+    fireEvent.click(screen.getByTestId("add-field"));
+    checkForDisablePreFillCheckox(props?.disableOnPrefill);
+    fireEvent.keyDown(screen.getByTestId("test-field-type"), { key: "ArrowDown", code: "ArrowDown" });
+    fireEvent.click(screen.getByTestId(`select-option-${props.questionType}`));
+    fireEvent.change(screen.getAllByRole("textbox")[0], { target: { value: props.identifier } });
+    fireEvent.change(screen.getAllByRole("textbox")[1], { target: { value: props.label } });
+    clickDisablePrefillBasedOnUserInput(props?.disableOnPrefill);
+    fireEvent.click(screen.getByTestId("field-add-save"));
+    await waitFor(() => {
+      expect(screen.queryByTestId(`field-${props.identifier}`)).toBeInTheDocument();
+    });
+  },
+  editQuestion: async (props: {
```

Comment:
I see similar code in here. We must reuse and add to the existing code if needed.

![image](https://github.com/user-attachments/assets/77afd20d-06bb-4d42-9505-c38e5bb2bb20)